src/shared/utilities/messages.ts

```diff
@@ -182,11 +183,114 @@ export async function showMessageWithCancel(
     message: string,
     timeout: Timeout,
     window: Window = globals.window
-): Promise<vscode.Progress<{ message?: string; increment?: number }>> {
+): Promise<vscode.Progress<ProgressEntry>> {
     const progressOptions = { location: vscode.ProgressLocation.Notification, title: message, cancellable: true }
     return showProgressWithTimeout(progressOptions, timeout, window)
 }
 
+/**
+ * The message ID that {@link CancellableMessageManager} uses.
+ * It uniquely identifies a specific message.
+ */
+export type CancellableMessageId = string
```

Comment:
The goal of this was to improve readability so that using this type in function args or return types was obvious to the user if it was the message id vs. another string.

Reply:
> The goal of this was to improve readability so that using this type in function args or return types was obvious to the user if it was the message id vs. another string.

I understand the intent, I just personally don't find it useful because the string is crafted by the caller. So the intent is misleading if that makes sense? In any case this is a nitpick so if it makes sense to you it's fine with me.

Reply:
After dropping the `idFromCwlInfo()` method in favor of using the util methods, I've dropped this since I don't see it as necessary anymore.